File: app/src/main/java/dev/olog/basil/presentation/main/di/MainActivityInjector.java
package dev.olog.basil.presentation.main.di;

import android.app.Activity;

import dagger.Binds;
import dagger.Module;
import dagger.android.ActivityKey;
import dagger.android.AndroidInjector;
import dagger.multibindings.IntoMap;
import dev.olog.basil.presentation.main.MainActivity;

@Module(subcomponents = {
        MainActivitySubComponent.class
})
public abstract class MainActivityInjector {

    @Binds
    @IntoMap
    @ActivityKey(MainActivity.class)
    abstract AndroidInjector.Factory<? extends Activity> injectorFactory(MainActivitySubComponent.Builder builder);

}

File: presentation/src/main/java/dev/olog/basil/presentation/main/di/MainActivityInjector.kt
package dev.olog.basil.presentation.main.di

import dagger.Binds
import dagger.Module
import dagger.android.AndroidInjector
import dagger.multibindings.ClassKey
import dagger.multibindings.IntoMap
import dev.olog.basil.presentation.main.MainActivity

@Module(subcomponents = [MainActivityComponent::class])
abstract class MainActivityInjector {

    @Binds
    @IntoMap
    @ClassKey(MainActivity::class)
    internal abstract fun bindActivity(factory: MainActivityComponent.Factory): AndroidInjector.Factory<*>

}

